ydb/core/tx/schemeshard/schemeshard__background_cleaning.cpp

```diff
@@ -24,36 +61,204 @@ NOperationQueue::EStartStatus TSchemeShard::StartBackgroundCleaning(const TPathI
 
     auto ctx = ActorContext();
     LOG_INFO_S(ctx, NKikimrServices::FLAT_TX_SCHEMESHARD, "RunBackgroundCleaning "
-        "for temp table# " << JoinPath({info->WorkingDir, info->Name})
-        << ", ownerId# " << info->OwnerActorId
+        "for temp dir# " << JoinPath({info->WorkingDir, info->Name})
+        << ", ownerId# " << info->TempDirOwnerActorId
         << ", next wakeup# " << BackgroundCleaningQueue->GetWakeupDelta()
         << ", rate# " << BackgroundCleaningQueue->GetRate()
         << ", in queue# " << BackgroundCleaningQueue->Size() << " cleaning events"
         << ", running# " << BackgroundCleaningQueue->RunningSize() << " cleaning events"
         << " at schemeshard " << TabletID());
 
-    auto txId = GetCachedTxId(ctx);
+    auto traverseResult = Traverse(info->WorkingDir, info->Name, this);
+
+    auto [stateIter, _] = BackgroundCleaningState.emplace(
+        pathId,
+        TBackgroundCleaningState {
+            {},
+            std::move(traverseResult.Dirs),
+            std::move(traverseResult.Objects)
+        });
+    auto& state = stateIter->second;
+
+    for (const auto& tablePathId : state.ObjectsToDrop) {
+        const auto txId = GetCachedTxId(ctx);
```

Comment:
`GetCachedTxId()` might return `InvalidTxId` -- for example, in the case when there are more then 100 items in `ObjectsToDrop`, cache will deplete on 101+ items.

Loop should check returned `txId` and stop when there is no tx-id work with.